### PR TITLE
[DOCS] Adds inference API end-to-end example

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds a tutorial to the `Search your data` section that describes how to use the Inference API with the OpenAI service to use semantic search.

### Preview

[Semantic search with the inference API](https://elasticsearch_103042.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/semantic-search-inference.html)

### CODE REVIEW

**Comment by szabosteve** [comment]
run elasticsearch-ci/docs

**Comment by szabosteve** [comment]
run elasticsearch-ci/docs

**Comment by serenachou** [comment]
We'd like to see a notebook on Inference API scenario 
* connecting to Hugging Face for embeddings
* connecting to OpenAI for embeddings 

Linking that https://www.elastic.co/search-labs/tutorials/integrations/hugging-face 
linking to https://www.elastic.co/search-labs/tutorials/integrations/open-ai

**Comment by szabosteve** [comment]
@serenachou I copied your comments to the tracking issue to make sure they are captured and easy to find even after this PR is merged.

File: docs/reference/search/search-your-data/semantic-search-inference.asciidoc [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+In this step, you load the data that you later use in the {infer} ingest
+pipeline to create embeddings from it.
+
+Use the `msmarco-passagetest2019-top1000` data set, which is a subset of the MS
+MARCO Passage Ranking data set. It consists of 200 queries, each accompanied by
+a list of relevant text passages. All unique passages, along with their IDs,
+have been extracted from that data set and compiled into a
+https://github.com/elastic/stack-docs/blob/main/docs/en/stack/ml/nlp/data/msmarco-passagetest2019-unique.tsv[tsv file].
+
+Download the file and upload it to your cluster using the
```
Comment (davidkyle) [inline docs/reference/search/search-your-data/semantic-search-inference.asciidoc:143]: I got a timeout uploading the file to the Data Visualizer. Let's create a smaller file for the purpose of this example. For readers following this example there's no need to process 60MB of data

File: docs/reference/search/search-your-data/semantic-search-inference.asciidoc [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+The mapping of the destination index - the index that contains the embeddings 
+that the model will create based on your input text - must be created. The 
+destination index must have a field with the <<dense-vector, `dense_vector`>> 
+field type to index the output of the OpenAI model.
+
+[source,console]
+--------------------------------------------------
+PUT openai-embeddings
+{
+  "mappings": {
```
Comment (davidkyle) [inline docs/reference/search/search-your-data/semantic-search-inference.asciidoc:65]: The dense_vector field has an [element_type](https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html#dense-vector-params) option which defaults to float. For these embeddings which are quite large at 1536 elements let's use the `element_type: byte` option as it requires much less memory

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**